src/serializer/ResidualHeapSerializer.js

```diff
@@ -1026,7 +1026,17 @@ export class ResidualHeapSerializer {
       invariant(serializedArg);
       args.push(serializedArg);
     }
-    return t.callExpression(this.preludeGenerator.memoizeReference("Symbol"), args);
+    // check if symbol value exists in the global symbol map, in that case we emit an invocation of System.for
+    // to look it up
+    let globalReg = this.realm.globalSymbolRegistry.find(e => e.$Symbol === val) !== undefined;
```

Comment:
I think you want to remove "!== undefined" since you are checking "globalReg === undefined". Otherwise, you should use globalReg as boolean.